src/Microsoft.SymbolManifestGenerator/SymbolManifestGenerator.cs

```diff
@@ -10,110 +10,140 @@
 using Microsoft.SymbolStore;
 using Microsoft.SymbolStore.KeyGenerators;
 
-namespace Microsoft.SymbolManifestGenerator
+namespace Microsoft.SymbolManifestGenerator;
+
+public static class SymbolManifestGenerator
 {
-    public static class SymbolManifestGenerator
+    private const int Private = 340;
+
+    public static bool GenerateManifest(ITracer tracer, DirectoryInfo dir, string manifestFileName, bool specialFilesRequireAdjacentRuntime = true)
```

Comment:
What scenario would we use specialFilesRequireAdjacentRuntime = false?

Reply:
.NET Core doesn't guarantee that all special files for a particular runtime module are in the same exact directory as the runtime module.  Specifically the cross arch verson of the long name dac/sos will be in a different directory (they share a grandparent directory with the coreclr.dll I think).

Reply:
Would we want that behavior always? I guess when do we need the flag to be true and when will we need it to be false

Reply:
I think it would be nice if for .NET Core we could have the special files in the same place  as the runtime module like as done in .NET Framework, but that sounds like a breaking change.

We need the flag to be true for .NET Framework because there are two runtime module files within the directory. One being `clr.dll` the other in a subdir `unopt/clr.dll`, and there are only special files for the root dir `clr.dll`. 
The symbol manifest generator has no assumptions on where runtime modules are in the directory, and iterates over all files and determines if a file is a runtime module if it can return valid ClrKeys
So what was happening was both `clr.dll` and `unopt/clr.dll` would generate entries for the same special files, when it really should only generate entries for special files according to the root `clr.dll`.
So the `flag = true` is the workaround to get a correct manifest for .Net Framework since we know that the special files will be in the same dir as the runtime module, whereas for .NET Core we dont have the guarantee that special files are in the same dir.